lib/ts/fetch.ts

```diff
@@ -200,22 +200,17 @@ export default class AuthHttpRequest {
                 // we read this here so that if there is a session expiry error, then we can compare this value (that caused the error) with the value after the request is sent.
                 // to avoid race conditions
                 const preRequestIdToken = await getIdRefreshToken(true);
-                let configWithAntiCsrf: RequestInit | undefined = config;
+                const clonedHeaders = new Headers(
+                    config !== undefined && config.headers !== undefined ? config.headers : url.headers
+                );
```

Comment:
The `url` object might have other params set too that might need to be copied over to the `config` object? Cause if someone is using that, then they are likely not using the `config` object.

So we will have to copy over all the properties to the `config` object

Reply:
1. As far as I can see from the spec, there are no other properties that'd be overwritten.
2. I don't see why we need this? If we do that, it might be better to copy and modify the request object instead of the init config.

Reply:
> As far as I can see from the spec, there are no other properties that'd be overwritten.

If this is true, then we don't need to copy over anything else.